packages/cli/src/commands/publish.test.ts

```diff
@@ -0,0 +1,232 @@
+    beforeEach(() => {
+      // Reset mocks
+      jest.resetAllMocks();
+
+      const nonSenseFileName = 'nonSenseFileName.txt';
+      const _deployDataLocalFileNames = [
+        `${packageName.replace(':', '_')}_${chainId}-${preset}.txt`,
+        `${packageName.replace(':', '_')}_${chainId}-${otherPreset}.txt`,
+        `${packageName.replace(':', '_')}_${otherChainId}-${preset}.txt`,
+        nonSenseFileName,
+      ];
+      // @ts-ignore
+      jest.spyOn(fs, 'readdir').mockResolvedValue(_deployDataLocalFileNames);
+      jest.spyOn(builder, 'copyPackage').mockImplementation(async () => {
+        return [];
+      });
+    });
+
+    it('should only find single deploy file on chainId and preset set', async () => {
+      await publish({
+        packageRef: packageName,
+        signer,
+        tags,
+        chainId,
+        preset,
+        quiet: true,
+        recursive: true,
+        overrides: {},
+      });
+
+      expect(copyPackage as jest.Mock).toHaveBeenCalledTimes(1);
+      expect((copyPackage as jest.Mock).mock.calls[0][0].packageRef).toEqual(packageName);
+      expect((copyPackage as jest.Mock).mock.calls[0][0].variant).toEqual(`${chainId}-${preset}`);
+    });
+
+    // Not sure if it's the expected behavior to match multiple deploy files on preset is empty
+    // But it's the current implementation
+    it('should find multiple deploy files on chainId set', async () => {
+      await publish({
+        packageRef: packageName,
+        signer,
+        tags,
+        chainId,
+        preset: '',
+        quiet: true,
+        recursive: true,
+        overrides: {},
+      });
+
+      expect(copyPackage as jest.Mock).toHaveBeenCalledTimes(2);
+      expect((copyPackage as jest.Mock).mock.calls[0][0].packageRef).toEqual(packageName);
+      expect((copyPackage as jest.Mock).mock.calls[0][0].variant).toEqual(`${chainId}-${preset}`);
+      expect((copyPackage as jest.Mock).mock.calls[1][0].packageRef).toEqual(packageName);
+      expect((copyPackage as jest.Mock).mock.calls[1][0].variant).toEqual(`${chainId}-${otherPreset}`);
+    });
+
+    // Not sure if it's the expected behavior to match multiple deploy files on chainId is zero
+    // But it's the current implementation
+    it('should find multiple deploy files on preset set', async () => {
+      await publish({
+        packageRef: packageName,
+        signer,
+        tags,
+        chainId: 0,
+        preset,
+        quiet: true,
+        recursive: true,
+        overrides: {},
+      });
+
+      expect(copyPackage as jest.Mock).toHaveBeenCalledTimes(2);
+      expect((copyPackage as jest.Mock).mock.calls[0][0].packageRef).toEqual(packageName);
+      expect((copyPackage as jest.Mock).mock.calls[0][0].variant).toEqual(`${chainId}-${preset}`);
+      expect((copyPackage as jest.Mock).mock.calls[1][0].packageRef).toEqual(packageName);
+      expect((copyPackage as jest.Mock).mock.calls[1][0].variant).toEqual(`${otherChainId}-${preset}`);
+    });
```

Comment:
It would be better to remove/skip these tests because they are written based on the current implementation of methods rather than the expected behaviour